napalm_ansible/modules/napalm_get_config.py

```
@@ -0,0 +1,267 @@
+        if module.params.get('optional_args') and module.params['optional_args'].get(param):
+            module.no_log_values.update(return_values(module.params['optional_args'].get(param)))
+
+    # allow host or hostname
+    provider['hostname'] = provider.get('hostname', None) or provider.get('host', None)
+    # allow local params to override provider
+    for param, pvalue in provider.items():
+        if module.params.get(param) is not False:
+            module.params[param] = module.params.get(param) or pvalue
+
+    hostname = module.params['hostname']
+    username = module.params['username']
+    dev_os = module.params['dev_os']
+    password = module.params['password']
+    timeout = module.params['timeout']
+    dest = module.params['dest']
+    type = module.params['type']
+    strip_comments = module.params['strip_comments']
+
+    argument_check = {'hostname': hostname, 'username': username, 'dev_os': dev_os}
+    for key, val in argument_check.items():
+        if val is None:
+            module.fail_json(msg=str(key) + " is required")
+
+    if module.params['optional_args'] is None:
+        optional_args = {}
+    else:
+        optional_args = module.params['optional_args']
+
+    try:
+        network_driver = get_network_driver(dev_os)
+        device = network_driver(hostname=hostname,
+                                username=username,
+                                password=password,
+                                timeout=timeout,
+                                optional_args=optional_args)
+        device.open()
+    except Exception as e:
+        module.fail_json(msg="cannot connect to device: " + str(e))
+
+    try:
+        config = device.get_config(retrieve=type)[type]
+        # if retrieved config is empty retrieve "running" type instead
+        if len(config) == 0:
+            config = device.get_config(retrieve="running")["running"]
```

I don't think we should do this...I think we should just have it default to 'running' but if you specify a type of config to be retrieved and that type fails, then we should just raise an exception.

---

fixed